node/src/components/contract_runtime/operations.rs

```diff
@@ -69,6 +69,18 @@ pub fn execute_finalized_block(
     // Create a new EngineState that reads from LMDB but only caches changes in memory.
     let scratch_state = engine_state.get_scratch_engine_state();
 
+    // Pay out block rewards
+    let proposer = *finalized_block.proposer();
+    state_root_hash = scratch_state.distribute_block_rewards(
```

Comment:
We should discuss whether we want to store the `ExecutionResult`s that represent this state change.